src/Symfony/Component/Console/Helper/Table.php

```diff
@@ -222,6 +224,37 @@ public function setColumnWidths(array $widths)
         return $this;
     }
 
+    /**
+     * Sets the maximum width of a column.
+     *
+     * @return $this
+     */
+    public function setColumnMaxWidth(int $columnIndex, int $width): self
+    {
+        if (!$this->output->getFormatter() instanceof WrappableOutputFormatterInterface) {
+            throw new \LogicException(sprintf('Setting a maximum column width is only supported when using a "%s" formatter, got "%s".', WrappableOutputFormatterInterface::class, \get_class($this->output->getFormatter())));
+        }
+
+        $this->columnMaxWidths[$columnIndex] = $width;
+
+        return $this;
+    }
+
+    /**
+     * Sets the maximum width of all columns.
+     *
+     * @return $this
+     */
+    public function setColumnMaxWidths(array $widths): self
```

Comment:
Do we really need this one? Can't we do the same with setColumnMaxWidth and keep a smaller public API?

Reply:
currently we have `setColumnWidth()` and `setColumnWidths()` to control min. width. We can do without `setColumnMaxWidths()` but might raise some questions (why cant i set max. widths at once?)

IMHO the ideal api is: `setColumnMinWidth + setColumnMaxWidth + setColumnWidth` where the latter sets both min. and max. (thus fixed). However that requires to rename the current `setColumnWidth` to `setColumnMinWidth` first. Not aiming for that  :)

Do you suggest to drop `setColumnMaxWidths` and add array support in `setColumnMaxWidth` or just drop it?